### PR TITLE
Added new arabic provider and added trailers to egybest

### PR DESCRIPTION
**Pull request by ImZaw**

### CODE REVIEW

**Comment by ImZaw** [comment]
HUH, Unresolved reference: addTrailer
what should it be then?

**Comment by ImZaw** [comment]
i think that should be there 
import com.lagradost.cloudstream3.LoadResponse.Companion.addTrailer
right?

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    override suspend fun getMainPage(): HomePageResponse {
+
+        val doc = app.get("$mainUrl/home", headers = mapOf("user-agent" to "MONKE")).document
+        val pages = arrayListOf<HomePageList>()
+        doc.select("section").not("section:contains(أختر وجهتك المفضلة)").not("section:contains(تم اضافته حديثاً)").apmap {
+            val name = it.select("span").html().replace("<em>.*| <i c.*".toRegex(), "")
+            val list = arrayListOf<SearchResponse>()
+            it.select("a").map {
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:52]: bruh, you can use the map to skip the variable assignment on the row above
If you want to skip stuff (line 53) you can use mapNotNull and return null :pray: 

Same goes for line 49 :pensive:

Comment (ImZaw) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:52]: zo skills 😜

Comment (ImZaw) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:52]: alright ill do it when i get on pc

**Comment by ImZaw** [comment]
now good?

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            null,
+            quality = getQualityFromString(quality)
+        )
+    }
+
+    override suspend fun getMainPage(): HomePageResponse {
+
+        val doc = app.get("$mainUrl/home", headers = mapOf("user-agent" to "MONKE")).document
+        val pages = arrayListOf<HomePageList>()
+        doc.select("section").not("section:contains(أختر وجهتك المفضلة)").not("section:contains(تم اضافته حديثاً)").apmap {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:49]: apmap is just a map so you should be able to just write val pages = doc.select.... if you remove pages.add

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                this.year = year
+                this.recommendations = recommendations
+                this.plot = synopsis
+                this.tags = tags
+                addTrailer(youtubeTrailer)
+            }
+        } else {
+            val episodes = ArrayList<Episode>()
+                doc.select("ul#eps li").map { episode ->
+                    episodes.add(
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:125]: do val episodes = doc.select("ul#eps li").map { Episode(...) } instead

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        app.get("$data").document.select("ul#download [aria-label=\"quality\"]").mapNotNull {
+            val name = if(it.select("span").text().contains("فائق السرعة")) "Fast Servers" else "Servers"
+            it.select("a").map { media ->
+                callback.invoke(
+                    ExtractorLink(
+                        source = this.name,
+                        name = name,
+                        url = media.attr("href"),
+                        referer = this.mainUrl,
+                        quality = media.text().getIntFromText()!!
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:161]: ?: Qualites.Unknown.value

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/EgyBestProvider.kt [hunk 5]
```diff
@@ -78,7 +79,8 @@ class EgyBestProvider : MainAPI() {
         val posterUrl = doc.select("div.movie_img a img")?.attr("src")
         val year = doc.select("div.movie_title h1 a")?.text()?.toIntOrNull()
         val title = doc.select("div.movie_title h1 span").text()
-
+        val youtubeTrailer = doc.select("div.play").attr("url")
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/EgyBestProvider.kt:82]: ?.

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        val year = doc.select("article ul:nth-child(1) li a").last()?.text()?.toIntOrNull()
+        val title = doc.select("title").text().split(" | ")[0]
+        val isMovie = title.contains("فيلم|حفلات|مسرحية".toRegex())
+        val youtubeTrailer = doc.select("iframe").attr("src")
+
+        val synopsis = doc.select("ul#details li:contains(لمحة) p").text()
+
+        val tags = doc.select("article ul").first()?.select("li")?.map { it.text() }
+
+        val recommendations = doc.select("ul#related li").mapNotNull { element ->
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:99]: you do actually not use null here?

**Review by LagradOst — APPROVED** [review]
Very minor code changes, mainly that you dont need to add to an arraylist if you use map 👍

**Comment by ImZaw** [comment]
@Blatzar

**Comment by ImZaw** [comment]
si now i really have skills, ill do them asap🧍

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        app.get("$data").document.select("ul#download [aria-label=\"quality\"]").mapNotNull {
+            val name = if(it.select("span").text().contains("فائق السرعة")) "Fast Servers" else "Servers"
+            it.select("a").map { media ->
+                callback.invoke(
+                    ExtractorLink(
+                        source = this.name,
+                        name = name,
+                        url = media.attr("href"),
+                        referer = this.mainUrl,
+                        quality = media.text().getIntFromText() ?: 0
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:157]: dont put 0, but Qualities.Unknown.Value

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        }
+    }
+
+    override suspend fun loadLinks(
+        data: String,
+        isCasting: Boolean,
+        subtitleCallback: (SubtitleFile) -> Unit,
+        callback: (ExtractorLink) -> Unit
+    ): Boolean {
+        app.get("$data").document.select("ul#download [aria-label=\"quality\"]").mapNotNull {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:148]: do forEach as you dont map anything

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+            val name = if(it.select("span").text().contains("فائق السرعة")) "Fast Servers" else "Servers"
+            it.select("a").map { media ->
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CimaNowProvider.kt:150]: forEach

**Review by LagradOst — APPROVED** [review]
👍

**Comment by LagradOst** [comment]
Ready to merge?

**Comment by ImZaw** [comment]
> Ready to merge?

Go ahead

**Pull request merged**